crates/ruff_python_literal/src/format.rs

```diff
@@ -282,17 +308,68 @@ fn parse_precision(text: &str) -> Result<(Option<usize>, &str), FormatSpecError>
     })
 }
 
+/// Parses a format part within a format spec
+fn parse_nested_format_part<'a>(
+    parts: &'a RefCell<Vec<FormatPart>>,
+    text: &'a str,
+) -> Result<&'a str, FormatSpecError> {
+    let mut parts = parts.borrow_mut();
+    let mut end_bracket_pos = None;
+    let mut left = String::new();
+
+    if text.is_empty() {
+        return Ok(text);
+    }
+
+    for (idx, c) in text.char_indices() {
+        if idx == 0 {
+            if c != '{' {
+                return Ok(text);
+            }
+        } else if c == '{' {
+            return Err(FormatSpecError::ReplacementNotAllowed);
+        } else if c == '}' {
+            end_bracket_pos = Some(idx);
+            break;
+        } else {
+            left.push(c);
+        }
+    }
+    if let Some(pos) = end_bracket_pos {
+        let (_, right) = text.split_at(pos);
+        let format_part = FormatString::parse_part_in_brackets(&left)
+            .map_err(|err| FormatSpecError::InvalidReplacement(err))?;
+        parts.push(format_part);
+        Ok(&right[1..])
+    } else {
+        Err(FormatSpecError::InvalidReplacement(
+            FormatParseError::MissingRightBracket,
+        ))
+    }
+}
+
 impl FormatSpec {
     pub fn parse(text: &str) -> Result<Self, FormatSpecError> {
+        let replacements = RefCell::new(vec![]);
         // get_integer in CPython
+        let text = parse_nested_format_part(&replacements, text)?;
```

Comment:
Would love suggestions here as this feels awkward.

Reply:
A bit tedious but I don't know that I have much better.